### PR TITLE
Port Lora example to Keras 3

### PR DESCRIPTION
**Pull request by mattdangerw**
Tensorflow backend for now just to preserve the GPU usage monitoring code.

### CODE REVIEW

File: examples/nlp/ipynb/parameter_efficient_finetuning_of_gpt2_with_lora.ipynb [hunk 1]
```diff
@@ -42,7 +42,9 @@
     "[GPT-2 model](https://keras.io/api/keras_nlp/models/gpt2/) and fine-tune\n",
     "it on the next token prediction task using LoRA. We will compare LoRA GPT-2\n",
     "with a fully fine-tuned GPT-2 in terms of the quality of the generated text,\n",
-    "training time and GPU memory usage."
+    "training time and GPU memory usage.\n",
+    "\n",
+    "Note: This example runs on the TensorFlow backend, purely because TensorFlow"
```
Comment (fchollet) [inline examples/nlp/ipynb/parameter_efficient_finetuning_of_gpt2_with_lora.ipynb:47]: Please regenerate the notebook, it seems there was a mixup (same in the md file)

File: examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py [hunk 2]
```diff
@@ -448,6 +456,8 @@ def call(self, inputs):
     # Change query dense layer.
     decoder_layer = lora_model.backbone.get_layer(f"transformer_layer_{layer_idx}")
     self_attention_layer = decoder_layer._self_attention_layer
+    # Allow mutation to Keras layer state.
+    self_attention_layer._tracker.locked = False
```
Comment (fchollet) [inline examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py:460]: Not optimal (not actionable here though). We should either make this a public API, or think of a better way to do this (preferable).

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/nlp/ipynb/parameter_efficient_finetuning_of_gpt2_with_lora.ipynb (hunk 1 above)
Comment (mattdangerw) [reply examples/nlp/ipynb/parameter_efficient_finetuning_of_gpt2_with_lora.ipynb]: done!

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**